infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/SessionManager.java

```diff
@@ -452,7 +463,17 @@ public Set<LogReplicationSession> getIncomingSessions() {
      * @return set of sessions for which this cluster is the sink
      */
     public Set<LogReplicationSession> getOutgoingSessions() {
-        return outgoingSessions;
+        if (replicationContext.getIsLeader().get()) {
+            return outgoingSessions;
+        }
+        List<CorfuStoreEntry<LogReplicationSession, ReplicationStatus, Message>> sessionsToStatus;
+        try (TxnContext txn = metadataManager.getTxnContext()) {
+            sessionsToStatus = txn.executeQuery(metadataManager.getStatusTable(), p -> true);
+            txn.commit();
+        }
+        Set<LogReplicationSession> sessionsToReturn = new HashSet<>();
+        sessionsToStatus.stream().filter(e -> e.getPayload().hasSourceStatus()).forEach(e -> sessionsToReturn.add(e.getKey()));
+        return sessionsToReturn;
```

Comment:
nit: we can avoid the temp var by using `sessionsToStatus.stream().filter(e -> e.getPayload().hasSourceStatus()).map(CorfuStoreEntry::getKey).collect(Collectors.toSet())`